roseau/load_flow/__init__.py

```diff
@@ -112,3 +113,28 @@
     "NegativeSequence",
     "ZeroSequence",
 ]
+
+global __rlf_deprecated_classes
+__rlf_deprecated_classes = set()
+
+
+def __getattr__(name: str) -> Any:
+    deprecated_classes = {"ConductorType": Material, "InsulatorType": Insulator}
+
+    if name in deprecated_classes:
+        from roseau.load_flow.utils._exceptions import find_stack_level
+
+        new_class = deprecated_classes[name]
+        if name not in __rlf_deprecated_classes:
+            import warnings
+
+            warnings.warn(
+                f"The {name} class is deprecated. Use {new_class.__name__} instead.",
+                stacklevel=find_stack_level(),
+            )
+        __rlf_deprecated_classes.add(name)
```

Comment:
Simpler implementation
```suggestion

def __getattr__(name: str) -> Any:
    deprecated_classes = {"ConductorType": Material, "InsulatorType": Insulator}

    if name in deprecated_classes:
        import warnings

        from roseau.load_flow.utils._exceptions import find_stack_level

        new_class = deprecated_classes[name]
        warnings.warn(
            f"The {name} class is deprecated. Use {new_class.__name__} instead.",
            stacklevel=find_stack_level(),
        )
        globals()[name] = new_class
```